File: Week_5_Data_structures/src/ListIntroduction2.java
import java.util.ArrayList;
import java.util.Arrays;

public class ListIntroduction2 {
    public static void main(String[] args) {

        //Create a list ('List A') which contains the following values
        //Apple, Avocado, Blueberries, Durian, Lychee
        ArrayList<String> list_A = new ArrayList<>();
        list_A.add("Apple");
        list_A.add("Avocado");
        list_A.add("Blueberries");
        list_A.add("Durian");
        list_A.add("Lychee");

        //Create a new list ('List B') with the values of List A
        ArrayList<String> list_B = new ArrayList<>();
        list_B.addAll(list_A);

        //Print out whether List A contains Durian or not
        System.out.println(list_A.contains("Durian"));

        //Remove Durian from List B
        list_B.remove("Durian");

        //Add Kiwifruit to List A after the 4th element
        list_A.add(5 , "Kiwifruit");

        //Compare the size of List A and List B
        System.out.println(list_A.equals(list_B));

        //Get the index of Avocado from List A
        System.out.println(list_A.indexOf("Avocado"));

        //Get the index of Durian from List B
        System.out.println(list_B.indexOf("Durian"));

        //Add Passion Fruit and Pomelo to List B in a single statement
        list_B.addAll(Arrays.asList("Passion Fruit" , "Pomelo"));
        System.out.println(list_B);

        //Print out the 3rd element from List A*/
        System.out.println(list_A.get(2));

    }
}

File: Project_phase_FE/DataStructures/src/ListIntroduction2.js
var listA = ["Apple", "Avocado", "Blueberries", "Durian", "Lychee"];

// Create a new list ('List B') with the values of List A
var listB = listA;

// Print out whether List A contains Durian or not
var boolean = listA.map(function(item){
    return item === "Durian";
});
console.log(boolean);

// Remove Durian from List B
listB.indexOf("Durian");
listB.splice(3, 1);
console.log(...listB);

// Add Kiwifruit to List A after the 4th element
listA.splice(4, 0, "Kiwifruit");
console.log(...listA);

// Compare the size of List A and List B
if (listA.length > listB.length){
    console.log("List A is bigger");
} else if (listA.length === listB.length) {
    console.log("Same size");
} else {
    console.log("List B is bigger");
}
console.log(...listA);
console.log(...listB);

// Get the index of Avocado from List A
console.log(listA.indexOf("Avocado"));

// Get the index of Durian from List B
console.log(listB.indexOf("Durian"));

// Add Passion Fruit and Pomelo to List B in a single statement
listB.push("Pomelo", "PassionFruit");
console.log(...listB);
console.log(...listA);

// Print out the 3rd element from List A
